### PR TITLE
Simplify ChunkTreeBuild

### PR DESCRIPTION
**Pull request by rynowak**
By making the tree more uniform and adding a root node, we simplify
the programming model inside the the builder.

### CODE REVIEW

File: src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs [hunk 1]
```diff
@@ -27,13 +30,13 @@ public void AddChunk(Chunk chunk, SyntaxTreeNode association, bool topLevel = fa
             chunk.Association = association;
 
             // If we're not in the middle of a parent chunk
-            if (_parentStack.Count == 0 || topLevel == true)
+            if (topLevel == true)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs]: Not sure why I had `topLevel == true`. Can just use this opportunity to simplify to `if (topLevel)` :smile:

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by rynowak** [comment]
I have one more idea I want to run by you tomorrow. Closing for now.

**Pull request closed without merging**